File: backend/src/main/java/eu/gillespie/bscthesis/request/ProveStatementRequest.java
package eu.gillespie.bscthesis.request;

import lombok.*;
import lombok.experimental.Accessors;
import java.util.List;

@Accessors(chain = true)
@Getter
@Setter
@ToString
@AllArgsConstructor
public class ProveStatementRequest {
    @NonNull String constructorSymbol;
    @NonNull StatementTreeVertex statementTree;
    @NonNull List<FunctionDefinition> functionDefinitions;

}

File: backend/src/main/java/eu/gillespie/bscthesis/request/ProveStatementRequest.kt
package eu.gillespie.bscthesis.request

import eu.gillespie.bscthesis.model.ConstructorDefinition

class ProveStatementRequest (
    var constructorDefinitions: List<ConstructorDefinition>,
    var statementTree: StatementTreeVertex,
    var functionDefinitions: List<FunctionDefinition>,
    val additionalConstraints: List<AdditionalConstraint>
)